sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -407,80 +366,25 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
+        this._context.connectionId,
+        this.name,
+        this.address,
+        senderError
+      );
+    }
     try {
       // Clears the token renewal timer. Closes the link and its session if they are open.
       // Removes the link and its session if they are present in rhea's cache.
       await this.closeLink();
-
-      // We should attempt to reopen only when the sender(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (senderError && !this.wasClosedPermanently) {
-        const translatedError = translate(senderError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Sender '%s' with address '%s' was not called. There " +
-              "was an accompanying error an it is retryable. This is a candidate for re-establishing " +
-              "the sender link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Sender '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the sender link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!this.wasClosedPermanently) {
-        shouldReopen = true;
-        log.error(
-          "[%s] close() method of Sender '%s' with address '%s' was not called. There " +
-            "was no accompanying error as well. This is a candidate for re-establishing " +
-            "the sender link.",
-          this._context.connectionId,
-          this.name,
-          this.address
-        );
-      } else {
-        const state: any = {
-          wasClosedPermanently: this.wasClosedPermanently,
-          senderError: senderError,
-          _sender: this.link
-        };
-        log.error(
-          "[%s] Something went wrong. State of sender '%s' with address '%s' is: %O",
-          this._context.connectionId,
-          this.name,
-          this.address,
-          state
-        );
-      }
-      if (shouldReopen) {
-        const senderOptions = this._createSenderOptions(
-          Constants.defaultOperationTimeoutInMs,
-          true
-        );
-        // shall retry as per the provided retryOptions if the error is a retryable error
-        // else bail out when the error is not retryable or the operation succeeds.
-        const config: RetryConfig<void> = {
-          operation: () => this.open(senderOptions),
-          connectionId: this._context.connectionId!,
-          operationType: RetryOperationType.senderLink,
-          retryOptions: this._retryOptions,
-          connectionHost: this._context.config.host
-        };
-        return await retry<void>(config);
-      }
     } catch (err) {
       log.error(
```

Comment:
This try/catch made sense before when did link recovery. Right now it applies to only the closeLink() call. This call internally catches error and silently logs it without re-throwing. All callers of `onDetached()` have their own catch where they log the error. So, do we really need this try/catch here at all?

Reply:
Can be removed since all the callers catch and log.

Reply:
Removed try-catch